src/util/KvAuth0.js

```diff
@@ -236,27 +272,31 @@ export default class KvAuth0 {
 			return Promise.resolve(this.mfaManagementToken);
 		}
 
+		// Fetch a new mfa management token
 		this[mfaTokenPromise] = new Promise((resolve, reject) => {
-			// Ensure browser clock is correct before fetching the token
-			syncDate().then(() => this[initMfaWebAuth]()).then(() => {
-				this.mfaWebAuth.checkSession({}, (err, result) => {
-					if (err) {
-						reject(err);
-					} else {
-						const { accessToken, expiresIn } = result;
-						// save access token as this.mfaEnrollToken
-						this.mfaManagementToken = accessToken;
-						// mfa management token expiration handling
-						if (expiresIn > 0) {
-							setTimeout(() => {
-								this.mfaManagementToken = '';
-							}, Number(expiresIn) * 1000);
-						}
-						// resolve with the mfa management token
+			// Initialize the web auth client
+			this[initWebAuth]()
+				// Parse the hash if it exists
+				.then(() => this[parseMfaHash]())
+				.then(() => {
+					if (this.mfaManagementToken) {
 						resolve(this.mfaManagementToken);
+					} else {
+						// If we still don't have a token, try to get one
+						// Ensure browser clock is correct before fetching the token
+						syncDate().then(() => {
+							// Store the current URL in session storage to redirect back to it after MFA
+							const state = storeRedirectState();
```

Comment:
Do we need to ensure this async method completes before calling `authorize` on the next line?

Reply:
Yes, good catch